bin-src/lib/uploadZip.js

```diff
@@ -0,0 +1,77 @@
+import retry from 'async-retry';
+import fs from 'fs-extra';
+import progress from 'progress-stream';
+
+const SENTINEL_SUCCESS_VALUE = 'OK';
+
+export async function uploadZip(ctx, path, url, contentLength, onProgress) {
+  let totalProgress = 0;
+
+  ctx.log.debug(`Uploading ${contentLength} bytes for '${path}' to '${url}'`);
+
+  return retry(
+    async () => {
+      const progressStream = progress();
+
+      progressStream.on('progress', ({ delta }) => {
+        totalProgress += delta;
+        onProgress(totalProgress);
+      });
+
+      const res = await ctx.http.fetch(
+        url,
+        {
+          method: 'PUT',
+          body: fs.createReadStream(path).pipe(progressStream),
+          headers: {
+            'content-type': 'application/zip',
+            'content-length': contentLength,
+            'cache-control': 'max-age=31536000',
```

Comment:
I'm sure I just copy-pasted, but I don't think `cache-control` is meaningful on a `PUT` here.

Reply:
Good catch!